features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/WindowInspector.kt

```diff
@@ -10,54 +10,56 @@ import android.annotation.SuppressLint
 import android.os.Build
 import android.view.View
 import android.view.inspector.WindowInspector
+import com.datadog.android.api.InternalLogger
 import java.lang.reflect.Field
 
 @SuppressLint("PrivateApi")
 internal object WindowInspector {
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    internal const val FAILED_TO_RETRIEVE_DECOR_VIEWS_ERROR_MESSAGE =
+        "SR WindowInspector failed to retrieve the decor views"
+
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_CLASS by lazy {
-        try {
-            return@lazy Class.forName("android.view.WindowManagerGlobal")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy Class.forName("android.view.WindowManagerGlobal")
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val GLOBAL_WM_INSTANCE by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getMethod("getInstance")?.invoke(null)
     }
 
-    @Suppress("SwallowedException", "TooGenericExceptionCaught")
+    @Suppress("SwallowedException", "TooGenericExceptionCaught", "UnsafeThirdPartyFunctionCall")
     private val VIEWS_FIELD by lazy {
-        try {
-            return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
-        } catch (e: Throwable) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // sdkLogger.errorWithTelemetry("WindowInspector failed to retrieve the decor views", e)
-        }
-        return@lazy null
+        return@lazy GLOBAL_WM_CLASS?.getDeclaredField("mViews")
     }
 
-    fun getGlobalWindowViews(): List<View> {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            WindowInspector.getGlobalWindowViews()
-        } else {
-            getGlobalWindowViewsLegacy(GLOBAL_WM_INSTANCE, VIEWS_FIELD)
+    @SuppressWarnings("TooGenericExceptionCaught")
```

Comment:
What are the exceptions we are expecting here? I'd rather we actually took the effort to know what the reflection calls might throw and only catch those, as catching any throwable could lead to keeping the JVM in a bad state (e.g.: catching an OOME)

Reply:
The idea is that is a global one because we are not very sure of the reflection exception in general but also because of the `WindowInspector.getGlobalWindowViews` which is a system call and calls `WindowManagerGlobal.getInstance().getWindowViews()` for which we do not have any information. That might throw also but we do not know what.

Reply:
Also this method is called every time we take a snapshot so it is very important to have it not to throw anything.